hazelcast/src/hazelcast/client/impl/RoundRobinLB.cpp

```diff
@@ -37,6 +37,8 @@ namespace hazelcast {
                 return members[++index % members.size()];
             }
 
+            RoundRobinLB::RoundRobinLB(const RoundRobinLB &rhs) : index(const_cast<RoundRobinLB &>(rhs).index.get()) {
```

Comment:
Instead of sharing the AtomicInt inside we are creating a duplicate here. 
This means that different places that copied RoundRobinLB will not be aware of each other. 
Not sure how much this affects the behaviour but this is a different behaviour from what we have in java.

I think we can achieve same behaviour by keeping shared_ptr<AtomicInt > for `index`